src/VisualStudio/Core/Def/Implementation/TableDataSource/Suppression/VisualStudioDiagnosticListTableCommandHandler.cs

```diff
@@ -0,0 +1,104 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.ComponentModel.Composition;
+using System.ComponentModel.Design;
+using Microsoft.VisualStudio.LanguageServices.Implementation.Suppression;
+using Microsoft.VisualStudio.Shell;
+
+namespace Microsoft.VisualStudio.LanguageServices.Implementation.TableDataSource
+{
+    [Export(typeof(VisualStudioDiagnosticListTableCommandHandler))]
+    internal partial class VisualStudioDiagnosticListTableCommandHandler
+    {
+        private readonly IVisualStudioSuppressionFixService _suppressionFixService;
+        private readonly IVisualStudioDiagnosticListSuppressionStateService _suppressionStateService;
+        
+        [ImportingConstructor]
+        public VisualStudioDiagnosticListTableCommandHandler(
+            IVisualStudioSuppressionFixService suppressionFixService,
+            IVisualStudioDiagnosticListSuppressionStateService suppressionStateService)
```

Comment:
IVisualStudioDiagnosticListSuppressionStateService is the one that has state, this one just registers the context menus and asks the state service for state when it gets a query status callback and IVisualStudioSuppressionFixService for bulk fix invocation from context menu.